src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -2053,24 +2057,69 @@ UpdateShardPlacement(PlacementUpdateEvent *placementUpdateEvent,
  * ExecuteCriticalCommandInSeparateTransaction runs a command in a separate
  * transaction that is commited right away. This is useful for things that you
  * don't want to rollback when the current transaction is rolled back.
+ * Set true to 'useExclusiveTransactionBlock' to initiate a BEGIN and COMMIT statements.
  */
 void
-ExecuteRebalancerCommandInSeparateTransaction(char *command)
+ExecuteRebalancerCommandInSeparateTransaction(char *command, bool
+											  useExclusiveTransactionBlock)
 {
 	int connectionFlag = FORCE_NEW_CONNECTION;
 	MultiConnection *connection = GetNodeConnection(connectionFlag, LocalHostName,
 													PostPortNumber);
 	StringInfo setApplicationName = makeStringInfo();
-	appendStringInfo(setApplicationName, "SET application_name TO %s",
+	appendStringInfo(setApplicationName, "SET LOCAL application_name TO %s;",
 					 CITUS_REBALANCER_NAME);
 
+	if (PropagateSessionSettingsForLoopbackConnection)
+	{
+		StringInfo setStatements = GetSetStatementsForNewConnections();
+		appendStringInfoString(setApplicationName, setStatements->data);
+	}
+
+	if (useExclusiveTransactionBlock)
+	{
+		ExecuteCriticalRemoteCommand(connection, "BEGIN;");
```

Comment:
I guess this is only needed for `DropOrphanedShardsInSeparateTransaction`
```suggestion
		ExecuteCriticalRemoteCommand(connection, "BEGIN;");
```

I think that's special case that shouldn't use this call anymore and should instead do its own thing. We don't really care about propagating these settings for that call anyway.

For the other case I think we shouldn't do our own transaction management and instead rely on `SendCommandListToWorkerOutsideTransaction`